### PR TITLE
format new message draft as text when not able to reply, updated unit tests

### PR DESCRIPTION
**Pull request by KyleCardwell**
… tests

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- formats a new message draft as text when there are no associations at all, or all associated triage groups are blocked.

### [MHV-52993](https://jira.devops.va.gov/browse/MHV-52993) - Blocked triage group API: Non-reassignable new message drafts ###

> Blocked triage group API: Non-re-assignable new message drafts
> 
> User Story: As a user, I want to quickly understand what actions I can take when I resume a new message draft and cannot reassign it to a different triage group so that I don't become confused and can quickly switch to another option for reaching my care teams.
> 
>  
> 
> Given:
> When:
> Then:
> 
> Feature Flag Y/N ? Y-One flag for all blocked triage group tickets (front end)
> Manual Testing Y/N ? Y
> Automated Testing Y/N ? Y
> Accessibility Testing Y/N ? Y
> UCD Validation? Y/N
> 
> Acceptance Criteria:
> AC1 Users blocked from all associations-system follows the current design (TBD)
> AC2 UCD has verified the design
> AC3
> AC4
> AC5
> 
> Links to UCD:
> Sketch Link:https://www.sketch.com/s/50ee7060-35ee-4493-ad26-aa819a77268b/a/uuid/536ADE47-C5F7-41C1-9224-CA0B9BD65873
> 
> https://www.sketch.com/s/50ee7060-35ee-4493-ad26-aa819a77268b#Version
> Other Design Notes
> 
> Architectural Notes: Associations should not be lost after 365 days.

## Testing done

- manual testing with and without triage groups to validate state of the new message draft form
- updated unit tests for each scenario

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop No Associations|Screenshot 2024-01-17 at 3 37 18 PM|Screenshot 2024-01-17 at 3 34 08 PM|
| Desktop All Blocked|Screenshot 2024-01-17 at 3 37 18 PM|Screenshot 2024-01-17 at 3 33 34 PM|

## What areas of the site does it impact?

- va.gov - Secure messaging

## Acceptance criteria

> AC1 Users blocked from all associations-system follows the current design (TBD)
> AC2 UCD has verified the design

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx [hunk 1]
```diff
@@ -661,47 +675,84 @@ const ComposeForm = props => {
...
+              />
+            )}
+            {mhvSecureMessagingBlockedTriageGroup1p0 &&
+              (noAssociations || allTriageGroupsBlocked ? (
+                <ViewOnlyDraftSection
+                  title={FormLabels.SUBJECT}
+                  body={subject}
+                />
+              ) : (
+                <></>
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/ComposeForm/ComposeForm.jsx:725]: same as comment above, see if `&&` is applicable here of `? : ` ternary operator

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadBody.jsx [hunk 2]
```diff
@@ -7,7 +7,7 @@ const MessageThreadBody = props => {
 
   const componentDecorator = (href, linkText) => (
     <a href={href} target="_blank" rel="noreferrer">
-      {linkText}
+      {`${linkText} (opens a new tab)`}
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadBody.jsx:10]: please confirm with UCD, I saw a not somewhere by a11y member where they mentioned that article `a` is not needed here in this expression

File: src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx [hunk 3]
```diff
@@ -687,6 +691,28 @@ describe('Compose form component', () => {
...
+        expect(draftSection.lastChild.textContent).to.equal(
+          customState.sm.threadDetails.drafts[0].subject,
+        );
+      }
+      if (index === 2) {
+        expect(draftSection.lastChild.textContent).to.equal(
+          customState.sm.threadDetails.drafts[0].messageBody,
+        );
+      }
+    });
```
Comment (oleksii-morgun-va) [inline src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx:715]: can you please elaborate on this statement, I am not sure I am following the use of index here

**Review by alex-morgun-1 — CHANGES_REQUESTED** [review]
Hey Kyle, great to see this blocked triage group work coming to conclusion. Very close. Left couple comments, some things may need to be tweaked

File: src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx (hunk 3 above)
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx:715]: I created one component that has a test-id of "view-only-draft-section" and I'm using that component to render the category, subject, and message body, so when I query that test-id, I get an array of 3.  I wanted to compare the rendered text to the text stored in the drafts[0] object, and each of the 3 elements is at a diferent index in the array.

File: src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadBody.jsx (hunk 2 above)
Comment (KyleCardwell) [reply src/applications/mhv/secure-messaging/components/MessageThread/MessageThreadBody.jsx:10]: reached out to UCD and the final decision is to say "(opens in new tab)".  I'll update it.

File: src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx (hunk 3 above)
Comment (oleksii-morgun-va) [reply src/applications/mhv/secure-messaging/tests/components/Compose/ComposeForm.unit.spec.jsx:715]: aha, okay, that makes sense now. Thanks

**Review by alex-morgun-1 — APPROVED** [review]
Approved, thank you for the follow up

**Pull request merged**